src/modules/fancyzones/editor/FancyZonesEditor/CanvasZone.xaml.cs

```diff
@@ -16,192 +17,333 @@ namespace FancyZonesEditor
 
-        private static readonly int _minZoneWidth = 64;
-        private static readonly int _minZoneHeight = 72;
-        private static int _zIndex = 0;
+        private CanvasLayoutModel model;
 
-        public CanvasZone()
+        private int zoneIndex;
+
+        public enum ResizeMode
         {
-            InitializeComponent();
-            Panel.SetZIndex(this, _zIndex++);
+            BottomEdge,
+            TopEdge,
+            BothEdges,
         }
 
-        private void Move(double xDelta, double yDelta)
+        private abstract class SnappyHelperBase
         {
-            Int32Rect rect = Model.Zones[ZoneIndex];
-            if (xDelta < 0)
-            {
-                xDelta = Math.Max(xDelta, -rect.X);
-            }
-            else if (xDelta > 0)
-            {
-                xDelta = Math.Min(xDelta, _settings.WorkArea.Width - rect.Width - rect.X);
-            }
+            public int ScreenW { get; private set; }
 
-            if (yDelta < 0)
-            {
-                yDelta = Math.Max(yDelta, -rect.Y);
-            }
-            else if (yDelta > 0)
+            protected List<int> Snaps { get; private set; }
+
+            protected int MinValue { get; private set; }
+
+            protected int MaxValue { get; private set; }
+
+            public int Position { get; protected set; }
+
+            public ResizeMode Mode { get; private set; }
+
+            /// <summary>
+            /// Initializes a new instance of the <see cref="SnappyHelperBase"/> class.
+            /// Just pass it the canvas arguments. Use mode
+            /// to tell it which edges of the existing masks to use when building its list
+            /// of snap points, and generally which edges to track. There will be two
+            /// SnappyHelpers, one for X-coordinates and one for
+            /// Y-coordinates, they work independently but share the same logic.
+            /// </summary>
+            /// <param name="zones">The list of rectangles describing all zones</param>
+            /// <param name="zoneIndex">The index of the zone to track</param>
+            /// <param name="isX"> Whether this is the X or Y SnappyHelper</param>
+            /// <param name="mode"> One of the three modes of operation (for example: tracking left/right/both edges)</param>
+            /// <param name="screenAxisSize"> The size of the screen in this (X or Y) dimension</param>
+            public SnappyHelperBase(IList<Int32Rect> zones, int zoneIndex, bool isX, ResizeMode mode, int screenAxisSize)
             {
-                yDelta = Math.Min(yDelta, _settings.WorkArea.Height - rect.Height - rect.Y);
-            }
+                int zone_position = isX ? zones[zoneIndex].X : zones[zoneIndex].Y;
+                int zone_axis_size = isX ? zones[zoneIndex].Width : zones[zoneIndex].Height;
+                int min_axis_size = isX ? MinZoneWidth : MinZoneHeight;
+                List<int> key_positions = new List<int>();
```

Comment:
Camel case this

Reply:
I agree, preserving the existing casing is our policy.